### PR TITLE
Feature/ja 98 ad requests list student service

### PR DESCRIPTION
**Pull request by skrawus**
Added needed logic of `ViewAdRequests` in `StudentService` 
Completed TODOs in `StudentController`
Edited `AdRequestListDto` in the way I needed it to be used in `ViewAdRequests` - the changes should not conflict with any other code parts.

### CODE REVIEW

File: TutorLizard.BusinessLogic/Services/StudentService.cs [hunk 1]
```diff
@@ -1,11 +1,35 @@
...
+    public async Task<StudentsAdRequestsResponse> ViewAdRequests(StudentsAdRequestsRequest request)
+    {
+        var studentId = request.StudentId;
+
+        var adRequests = await _adRequestRepository.GetAll()
+            .Where(ar => ar.StudentId == studentId && ar.DateCreated != null)
+            .ToListAsync();
+
+        var adRequestsListDtos = adRequests
+            .Select(ar => new AdRequestsListDto())
```
Comment (Zjyslav) [inline TutorLizard.BusinessLogic/Services/StudentService.cs]: You create an empty `AdRequestListDto`.

File: TutorLizard.Web/Controllers/StudentController.cs [hunk 2]
```diff
@@ -61,16 +61,11 @@ public IActionResult AdRequests(IFormCollection buttonAction)
...
             StudentsAdRequestsRequest request = new(studentId);
 
-            StudentsAdRequestsResponse response = new()
-            {
-                //data for tests only
-                AdRequests = [new AdRequestsListDto(1, 1, 1, false, "xyz", "yxz", false)]
-            };
+            StudentsAdRequestsResponse response = await _studentService.ViewAdRequests(request);
 
             if (!string.IsNullOrWhiteSpace(buttonAction["btnRequest"]))
```
Comment (Zjyslav) [inline TutorLizard.Web/Controllers/StudentController.cs]: I don't understand this if statement and use of `IFormCollection` in general. This action is a GET, it's not a response to a form being sent.

**Review by Zjyslav — CHANGES_REQUESTED** [review]

File: TutorLizard.BusinessLogic/Services/StudentService.cs (hunk 1 above)
Comment (skrawus) [reply TutorLizard.BusinessLogic/Services/StudentService.cs]: Fixed

File: TutorLizard.Web/Controllers/StudentController.cs (hunk 2 above)
Comment (skrawus) [reply TutorLizard.Web/Controllers/StudentController.cs]: Deleted

**Review by Zjyslav — APPROVED** [review]
It works.

**Pull request merged**